src/app/marketplace/marketplace-cancel-purchase-button/marketplace-cancel-purchase-button.component.ts

```diff
@@ -48,11 +54,11 @@ export class MarketplaceCancelPurchaseButtonComponent implements OnInit, OnDestr
   }
 
   get isDisabled(): boolean {
-    if (!this.dataProduct.blockchainState) {
+    if (!this.blockchainDataProduct) {
```

Comment:
wouldn't be easier to use: `return this.blockchainDataProduct && this.blockchainDataProduct.disabled`?